modules/core/04-channel/types/msgs.go

```diff
@@ -499,18 +499,57 @@ var _ sdk.Msg = &MsgChannelUpgradeInit{}
 
 // NewMsgChannelUpgradeInit constructs a new MsgChannelUpgradeInit
 // nolint:interfacer
-func NewMsgChannelUpgradeInit() *MsgChannelUpgradeInit {
-	return &MsgChannelUpgradeInit{}
+func NewMsgChannelUpgradeInit(
+	portID, channelID string,
+	proposedUpgradeChannel Channel,
+	counterpartyTimeoutHeight clienttypes.Height,
+	counterpartyTimeoutTimestamp uint64,
+	signer string,
+) *MsgChannelUpgradeInit {
+	return &MsgChannelUpgradeInit{
+		PortId:                       portID,
+		ChannelId:                    channelID,
+		ProposedUpgradeChannel:       proposedUpgradeChannel,
+		CounterpartyTimeoutHeight:    counterpartyTimeoutHeight,
+		CounterpartyTimeoutTimestamp: counterpartyTimeoutTimestamp,
+		Signer:                       signer,
+	}
 }
 
 // ValidateBasic implements sdk.Msg
 func (msg MsgChannelUpgradeInit) ValidateBasic() error {
-	return errors.New("error method not implemented")
+	if err := host.PortIdentifierValidator(msg.PortId); err != nil {
+		return sdkerrors.Wrap(err, "invalid port ID")
+	}
+
+	if !IsValidChannelID(msg.ChannelId) {
+		return ErrInvalidChannelIdentifier
+	}
+	if msg.ProposedUpgradeChannel.State != INITUPGRADE {
+		return sdkerrors.Wrapf(ErrInvalidChannelState, "expected: %s, got: %s", INITUPGRADE, msg.ProposedUpgradeChannel.State)
+	}
+	if msg.CounterpartyTimeoutHeight.IsZero() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidHeight, "timeout height must be non-zero")
+	}
+
```

Comment:
This should be an OR

If Timeout Height is nonzero OR timeouttimestamp is nonzero this is a valid message.

You have implemented an AND

Replicate the logic for timeout fields here:

https://github.com/cosmos/ibc-go/blob/main/modules/core/04-channel/types/packet.go#L106